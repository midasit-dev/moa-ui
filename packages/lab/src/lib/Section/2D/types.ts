/**
 * Coord2D is a type that is used to define the coordinates of a shape.
 * @var x is the x-coordinate of the shape.
 * @var y is the y-coordinate of the shape.
 */
export type Coord2D = {
	x: number;
	y: number;
};

/**
 * Dimension2D is a type that is used to define the dimension of a shape.
 * @var width is the width of the canvas.
 * @var height is the height of the canvas.
 */
export type Dimension2D = {
	width: number;
	height: number;
}

/**
 * CanvasDimension2D is a type that is used to define the dimension of a shape.
 * @var Dimension2D is the dimension of the canvas.
 */
export type CanvasDimension2D = Dimension2D | [number, number];

/**
 * StartCoordinate2D is a type that is used to define the starting coordinates of a shape.
 * @var Coord2D is the coordinates of the shape.
 */
export type StartCoordinate2D = Coord2D | [number, number];

/**
 * Canvas is a type that is used to define the properties of a shape.
 * @var padding is the padding of the canvas. (default: 0, 1 -> 8px)
 * @var background is the background color of the canvas.
 * @var dimension is the dimension of the canvas.
 */
export interface Canvas {
	background?: string | null;
	dimension?: CanvasDimension2D;
};

/**
 * Shape is a type that is used to define the properties of a shape.
 * @var startCoords is the starting coordinates of the shape.
 * @var fill is the fill color of the shape. (background color)
 * @var stroke is the stroke color of the shape.
 * @var strokeWeight is the stroke weight of the shape.
 */
export interface Shape {
	startCoords?: StartCoordinate2D;
	fill?: string;
	stroke?: string;
	strokeWeight?: number;
}

/**
 * DimensionLine is a type that is used to define the properties of a shape. 치수선!
 * @var offset is the offset of the dimension line.
 * @var lineExtension is the line extension of the dimension line.
 * @var lineExtensionAngle is the line extension angle of the dimension line.
 * @var lineColor is the line color of the dimension line.
 * @var lineWeight is the line weight of the dimension line.
 * @var text is the text of the dimension line.
 * @var textColor is the text color of the dimension line.
 * @var textSize is the text size of the dimension line.
 * @var textOffset is the text offset of the dimension line.
 */
export interface DimensionLine {
	offset?: number;
	lineExtension?: number;
	lineExtensionAngle?: number;
	lineColor?: string;
	lineWeight?: number;
	text?: string | null;
	textColor?: string;
	textSize?: number;
	textOffset?: number;
}

/**
 * PreDefined is a type that is used to define the properties of a shape.
 * @var dimensionLine is the dimension line of the SolidRectangle. (b, h)
 * @var shape is the shape of the SolidRectangle.
 * @var b is width of the shape.
 * @var h is height of the shape.
 */
export interface SolidRectangleProps {
	canvas?: Canvas;
	shape?: Shape;
	dimensionLine?: {
		b?: DimensionLine;
		h?: DimensionLine;
	};

	b: number;
	h: number;
}

/**
 * HSectionProps is a type that is used to define the properties of a shape.
 * @var dimensionLine is the dimension line of the HSection.
 * @var shape is the shape of the HSection.
 * @var h is the height of the shape.
 * @var tw is the width of the web.
 * @var b1 is the width of the top flange.
 * @var tf1 is the thickness of the top flange.
 * @var b2 is the width of the bottom flange.
 * @var tf2 is the thickness of the bottom flange.
 * @var r1 is the radius of the top flange.
 * @var r2 is the radius of the bottom flange.
 */
export interface HSectionProps {
	canvas?: Canvas;
	shape?: Shape;
	dimensionLine?: {
		h?: DimensionLine;
		tw?: DimensionLine;
		b1?: DimensionLine;
		tf1?: DimensionLine;
		b2?: DimensionLine;
		tf2?: DimensionLine;
		r1?: DimensionLine;
		r2?: DimensionLine;
	};

	h: number;
	tw: number;
	b1: number;
	tf1: number;
	b2: number;
	tf2: number;
	r1: number;
	r2: number;
}