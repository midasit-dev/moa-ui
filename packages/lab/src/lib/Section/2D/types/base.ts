/**
 * Vertex2D is a type that is used to define the coordinates of a shape.
 * @var x is the x-coordinate of the shape.
 * @var y is the y-coordinate of the shape.
 */
export type Vertex2D = {
	x: number;
	y: number;
};

/**
 * Dimension2D is a type that is used to define the dimension of a shape.
 * @var width is the width of the canvas.
 * @var height is the height of the canvas.
 */
export type Dimension2D = {
	width: number;
	height: number;
}

/**
 * CanvasDimension2D is a type that is used to define the dimension of a shape.
 * @var Dimension2D is the dimension of the canvas.
 */
export type CanvasDimension2D = Dimension2D | [number, number];

/**
 * TranslateVertex2D is a type that is used to define the starting coordinates of a shape.
 * @var Vertex2D is the coordinates of the shape.
 */
export type TranslateVertex2D = Vertex2D | [number, number];

/**
 * Canvas is a type that is used to define the properties of a shape.
 * @var background is the background color of the canvas.
 * @var dimension is the dimension of the canvas.
 * @var translateCoords is the starting coordinates of the shape.
 * @var autoScale is the auto scale of the canvas.
 * @var scale is the scale of the canvas.
 * @var rotate is the rotate of the canvas. (in degrees, 0 - 360)
 * @var guideLine is the guide line of the canvas.
 */
export interface Canvas {
	background?: string | null;
	dimension?: CanvasDimension2D;
	translateCoords?: TranslateVertex2D;
	autoScale?: boolean;
	scale?: number;
	rotate?: number;
	guideLine?: boolean;
};

/**
 * Shape is a type that is used to define the properties of a shape.
 * @var fill is the fill color of the shape. (background color)
 * @var stroke is the stroke color of the shape.
 * @var strokeWeight is the stroke weight of the shape.
 */
export interface Shape {
	fill?: string;
	stroke?: string;
	strokeWeight?: number;
}

/**
 * DimensionLine is a type that is used to define the properties of a shape. 치수선!
 * @var offset is the offset of the dimension line.
 * @var lineExtension is the line extension of the dimension line.
 * @var lineExtensionAngle is the line extension angle of the dimension line.
 * @var lineColor is the line color of the dimension line.
 * @var lineWeight is the line weight of the dimension line.
 * @var text is the text of the dimension line.
 * @var textColor is the text color of the dimension line.
 * @var textSize is the text size of the dimension line.
 * @var textOffset is the text offset of the dimension line.
 */
export interface DimensionLine {
	offset?: number;
	lineExtension?: number;
	lineExtensionAngle?: number;
	lineColor?: string;
	lineWeight?: number;
	text?: string | null;
	textColor?: string;
	textSize?: number;
	textOffset?: number | null;
}

/**
 * LeaderLine is a type that is used to define the properties of a shape.
 * @var halfLength is the half length of the leader line.
 * @var lineColor is the line color of the leader line.
 * @var lineWeight is the line weight of the leader line.
 * @var text is the text of the leader line.
 * @var textColor is the text color of the leader line.
 * @var textSize is the text size of the leader line.
 */
export interface LeaderLine {
	halfLength?: number;
	lineColor?: string;
	lineWeight?: number;
	text?: string | null;
	textColor?: string;
	textSize?: number;
}

/**
 * SuperProps is a type that is used to define the properties of a section.
 * @var canvas is the canvas of the shape.
 */
export interface SuperProps {
	canvas?: Canvas;
}