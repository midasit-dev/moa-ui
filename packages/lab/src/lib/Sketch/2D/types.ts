/**
 * Coord2D is a type that is used to define the coordinates of a shape.
 * @var x is the x-coordinate of the shape.
 * @var y is the y-coordinate of the shape.
 */
export type Coord2D = {
	x: number;
	y: number;
};

/**
 * Dimension2D is a type that is used to define the dimension of a shape.
 * @var width is the width of the canvas.
 * @var height is the height of the canvas.
 */
export type Dimension2D = {
	width: number;
	height: number;
}

/**
 * CanvasDimension2D is a type that is used to define the dimension of a shape.
 * @var Dimension2D is the dimension of the canvas.
 */
export type CanvasDimension2D = Dimension2D | [number, number];

/**
 * StartCoordinate2D is a type that is used to define the starting coordinates of a shape.
 * @var Coord2D is the coordinates of the shape.
 */
export type StartCoordinate2D = Coord2D | [number, number];

/**
 * Canvas is a type that is used to define the properties of a shape.
 * @var background is the background color of the canvas.
 * @var dimension is the dimension of the canvas.
 */
export type Canvas = {
	background?: string;
	dimension?: CanvasDimension2D;
};

/**
 * Shape is a type that is used to define the properties of a shape.
 * @var startCoords is the starting coordinates of the shape.
 * @var fill is the fill color of the shape. (background color)
 * @var stroke is the stroke color of the shape.
 * @var strokeWeight is the stroke weight of the shape.
 */
export type Shape = {
	startCoords?: StartCoordinate2D;
	fill?: string;
	stroke?: string;
	strokeWeight?: number;
}

/**
 * UserDefined is a type that is used to define the properties of a shape.
 * @var canvasDim is the dimension of the canvas.
 * @var startCoords is the starting coordinates of the shape.
 */
export interface UserDefined {
	canvas?: Canvas;
	shape?: Shape;
}

/**
 * PreDefined is a type that is used to define the properties of a shape.
 * @var b is width of the shape.
 * @var h is height of the shape.
 */
export interface UserDefinedBox extends UserDefined {
	b: number;
	h: number;
}

// export interface UserDefinedH extends UserDefined {
// 	r: number;
// }