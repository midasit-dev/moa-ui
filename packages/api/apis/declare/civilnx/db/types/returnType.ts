export default interface ResponseType<BodyType> {
  [key: string]: BodyType;
}
