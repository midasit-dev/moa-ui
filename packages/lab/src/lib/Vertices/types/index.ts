export * from './props';
export * from './vertex';