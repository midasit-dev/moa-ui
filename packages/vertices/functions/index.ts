export * from './declare/to-vertices';