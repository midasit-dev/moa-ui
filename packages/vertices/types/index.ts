export * from './declare/props';
export * from './declare/vertex';