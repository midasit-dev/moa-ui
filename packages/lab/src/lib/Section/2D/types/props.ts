import { SuperProps, Canvas, Shape, DimensionLine } from '@lablib/Section/2D/types/base';

export interface FlexProps {
	canvas?: 					Canvas;
	solidRectangle?: 	Omit<SolidRectangleProps, 'canvas'>;
	hSection?: 				Omit<HSectionProps, 'canvas'>;
}

/**
 * PreDefined is a type that is used to define the properties of a shape.
 * @var referLine is the reference line of the shape.
 * @var shape is the shape of the SolidRectangle.
 * @var b is width of the shape.
 * @var h is height of the shape.
 */
export interface SolidRectangleProps extends SuperProps {
	shape?: Shape;
	referLine?: {
		b?: DimensionLine;
		h?: DimensionLine;
	};

	b: number;
	h: number;
}

/**
 * HSectionProps is a type that is used to define the properties of a shape.
 * @var referLine is the reference line of the shape.
 * @var shape is the shape of the HSection.
 * @var h is the height of the shape.
 * @var tw is the width of the web.
 * @var b1 is the width of the top flange.
 * @var tf1 is the thickness of the top flange.
 * @var b2 is the width of the bottom flange.
 * @var tf2 is the thickness of the bottom flange.
 * @var r1 is the radius of the top flange.
 * @var r2 is the radius of the bottom flange.
 */
export interface HSectionProps extends SuperProps {
	shape?: Shape;
	referLine?: {
		h?: DimensionLine;
		tw?: DimensionLine;
		b1?: DimensionLine;
		tf1?: DimensionLine;
		b2?: DimensionLine;
		tf2?: DimensionLine;
		r1?: DimensionLine;
		r2?: DimensionLine;
	};

	h: number;
	tw: number;
	b1: number;
	tf1: number;
	b2: number;
	tf2: number;
	r1: number;
	r2: number;
}