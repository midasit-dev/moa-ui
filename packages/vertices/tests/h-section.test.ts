import { toVertices } from "../functions";
import { ConvertProps, Vertex2D } from '../types';

describe('to-vertices.ts test', () => {
	test('h-section.ts test', () => {
		const vertices: Vertex2D[][] | Error = toVertices({
      type: "HSection",
      properties: {
        h: 300,
        tw: 30,
        b1: 250,
        tf1: 50,
        b2: 250,
        tf2: 50,
        r1: 10,
        r2: 10,
      },
    });

		if (vertices instanceof Error) {
			fail(`Function returned an error: ${vertices.message}`);
		} else {
			expect(vertices).toStrictEqual([
				[
					{ "x": -125, 	"y": 150 	},
					{ "x": 125, 	"y": 150 	},
					{ "x": 125, 	"y": 110 	},
					{ "x": 115, 	"y": 100 	},
					{ "x": 25, 		"y": 100 	},
					{ "x": 15, 		"y": 90 	},
					{ "x": 15, 		"y": -90 	},
					{ "x": 25, 		"y": -100 },
					{ "x": 115, 	"y": -100 },
					{ "x": 125, 	"y": -110 },
					{ "x": 125, 	"y": -150 },
					{ "x": -125, 	"y": -150 },
					{ "x": -125, 	"y": -110 },
					{ "x": -115, 	"y": -100 },
					{ "x": -25, 	"y": -100 },
					{ "x": -15, 	"y": -90 	},
					{ "x": -15, 	"y": 90 	},
					{ "x": -25, 	"y": 100 	},
					{ "x": -115, 	"y": 100 	},
					{ "x": -125, 	"y": 110 	}
				]
			]);
		}
	});
});
