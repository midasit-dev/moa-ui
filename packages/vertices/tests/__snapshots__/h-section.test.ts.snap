// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions/h-section.ts H-300x250x30x50 1`] = `
[
  [
    {
      "x": -125,
      "y": 150,
    },
    {
      "x": 125,
      "y": 150,
    },
    {
      "x": 125,
      "y": 110,
    },
    {
      "x": 115,
      "y": 100,
    },
    {
      "x": 25,
      "y": 100,
    },
    {
      "x": 15,
      "y": 90,
    },
    {
      "x": 15,
      "y": -90,
    },
    {
      "x": 25,
      "y": -100,
    },
    {
      "x": 115,
      "y": -100,
    },
    {
      "x": 125,
      "y": -110,
    },
    {
      "x": 125,
      "y": -150,
    },
    {
      "x": -125,
      "y": -150,
    },
    {
      "x": -125,
      "y": -110,
    },
    {
      "x": -115,
      "y": -100,
    },
    {
      "x": -25,
      "y": -100,
    },
    {
      "x": -15,
      "y": -90,
    },
    {
      "x": -15,
      "y": 90,
    },
    {
      "x": -25,
      "y": 100,
    },
    {
      "x": -115,
      "y": 100,
    },
    {
      "x": -125,
      "y": 110,
    },
  ],
]
`;
